client/webserver/site/src/html/bodybuilder.tmpl

```diff
@@ -1,9 +1,27 @@
+{{define "top"}}
+<!DOCTYPE html>
+<html lang="en">
+<head data-turbolinks-eval="false">
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <!-- The above 2 meta tags *must* come first in the head; any other head content must come *after* these tags -->
+  <link rel="icon" href="/img/favicon.png?v=yHh3NF">
+  <meta name="description" content="Decred DEX Client Web Portal">
+  <title>{{.Title}}</title>
+  <link href="/css/style.css?v=1" rel="stylesheet">
+</head>
+<body {{if .UserInfo.DarkMode}} class="dark"{{end}}>
+  <div class="poke-note px-2 fs14 d-flex align-items-center" id="pokeNote"><span>Here is a poke note.</span></div>
+  {{template "header" .}}
+{{end}}
+
 {{define "header"}}
 <header id=header class="maintop">
   {{$authed := .UserInfo.Authed}}
+  {{$marketsReady := and $authed (gt (len .UserInfo.Exchanges) 0)}}
   <div data-pagelink="" class="logo"></div>
   <div class="mainlinks fs18 offwhite">
-    <span data-pagelink="markets" class="demi hoverbright">Markets</span>
+    <span data-pagelink="markets" class="demi hoverbright{{if not $marketsReady}} d-hide{{end}}" id="marketsMenuEntry">Markets</span>
```

Comment:
This is fine, but it's not being shown on successful login or registration.

Reply:
Successful login calls `await app.fetchUser()` and `app.setLogged(true)`, so if the user has 1+ exchanges, the link would show.

Successful registration calls `await app.fetchUser()` but not `app.setLogged(true)` and it's `app.setLogged` that modifies the menu items visibility.

I'm thinking `app.setLogged` should be called whenever user info is refreshed, so that any changes to the user's authentication state or exchanges can be taken into account and the menu items visibility updated accordingly. That would mean login needs not call `app.setLogged(true)` after calling `app.fetchUser()`.

Reply:
Yes, `setLogged` is no longer just setting things based on the user being logged in, so that function name is outdated. 

> I'm thinking app.setLogged should be called whenever user info is refreshed

I agree. Wanna just run `setLogged` (with a better name) in `fetchUser`?

Reply:
Ah, I think a function rename would do the trick. Seeing as the function just updates menu items visibility currently, there shouldn't be a need to call it as part of `fetchUser` whenever `fetchUser` is called. Can limit calls to just where it is necessary, when user inits the app, logs in or connects a DEX.

Reply:
Okay, I admit. It reads better having `fetchUser` call `updateMenuItemsDisplay` since user info may have changed.